### DISCUSSION TITLE
The first trial suggestion starts with the same parameters across multiple computers

### DISCUSSION THREAD

**Discussion by matheus695p** [optuna/optuna#4715] [Q&A] [answered]
### Don't use GitHub Issues to ask support questions.

The first trial suggestion starts with the same parameters when running in distributed. Ex:

- I am using TPESampler just giving it the random seed.
- I'm connecting to a SQL database.

Ex1:
Running in azure compute clusters with 3 VM's in parallel:
trial 0 1 and 2 always start with the same configuration and deliver the same value.
E2:
Running locally from 3 terminals:
trial 0 1 and 2 always start with the same configuration and deliver the same value.

Do you know why this could be happening? I assume that TPESampler delivers random search in the first trials, but it is not doing so, it is delivering the same configuration to all machines.
Do you know any workaround to distribute the configurations restricting the must be always different ?

**Comment by matheus695p** [comment]
Fixed using constant_liar = True in the TPESampler, but do you know what is the root cause ?

**Comment by nzw0301** [comment]
Since we cannot see the implementation, I'm not sure my answer is correct, but I suppose you are using the same random seed among `TPESampler` instances for every process. Could you try the different random seed values depending on the processes?

**Reply by matheus695p** [reply]
Thanks, is not this a reproducibility problem ?, how do i mange it if i don't have the same random state ?

**Reply by nzw0301** [reply]
As explained in https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-obtain-reproducible-optimization-results, Optuna cannot give reproducible results if distributed training is used.

**Reply by matheus695p** [reply, marked as answer]
got it thank you very much !
